### PR TITLE
Add E2E tests

### PR DESCRIPTION
**Pull request by simonjaeger**
Adding E2E tests for events and alarms. Adds the ability to create ACI's on-demand for tests, improvements to the TestEventProcessor, direct connection to Azure IoT Hub to make assertions on the payload, etc. All tested in a copy of pipeline.

### CODE REVIEW

**Review by peterbryntesson — APPROVED** [review]
LGTM!

**Comment by simonjaeger** [comment]
Investigating SDL error (credential scanning).

File: e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs [hunk 1]
```diff
@@ -589,5 +649,397 @@ private static PrivateKeyFile GetPrivateSshKey(IIoTPlatformTestContext context)
                 exception = exception.InnerException;
             }
         }
+
+        public static string PublishedNodesJson(this IIoTStandaloneTestContext context, uint port, string writerId, JArray opcEvents) {
```
[the remark this answers is not in the document]
Comment (koepalex) [reply e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs:653]: valid for all new methods of this class

[the remark this answers is not in the document]
Comment (simonjaeger) [reply e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs:660]: Fixed!

**Review by koepalex — APPROVED** [review]
:shipit:

**Pull request merged**